internal/common/baseclient/iito.go

```diff
@@ -6,14 +6,14 @@ import "github.com/SOMAS2020/SOMAS2020/internal/common/shared"
 // This information is fed to OfferGifts of all other clients.
 // COMPULSORY, you need to implement this method
 func (c *BaseClient) GetGiftRequests() []shared.GiftRequest {
-	// FIXME: Make this the appropriate length, the loop below doesn't run otherwise.
-	// Can't test server until this is fixed.
 	requests := []shared.GiftRequest{}
-	for fromTeam := range requests {
-		if c.clientGameState.ClientInfo.LifeStatus == shared.Critical {
-			requests[fromTeam] = shared.GiftRequest{RequestFrom: shared.ClientID(fromTeam), RequestAmount: 100.0}
+
+	// FIXME: Is this the right way to get the client statuses?
+	for team, status := range c.clientGameState.ClientLifeStatuses {
+		if status == shared.Critical {
+			requests[team] = shared.GiftRequest{RequestFrom: shared.ClientID(team), RequestAmount: 100.0}
```

Comment:
since this is the base client and this should be overridden it doesnt matter too much. But does this request resources from dying islands?

Reply:
Solved in latest update.